tools/stub_service/src/schema_registry/service.rs

```diff
@@ -0,0 +1,318 @@
+pub struct Service<C>
+where
+    C: ManagedClient + Clone + Send + Sync + 'static,
+    C::PubReceiver: Send + Sync + 'static,
+{
+    schemas: Arc<Mutex<HashMap<String, BTreeSet<Schema>>>>,
+    get_command_executor: service_gen::GetCommandExecutor<C>,
+    put_command_executor: service_gen::PutCommandExecutor<C>,
+    service_output_manager: ServiceOutputManager,
+}
+
+impl<C> Service<C>
+where
+    C: ManagedClient + Clone + Send + Sync + 'static,
+    C::PubReceiver: Send + Sync + 'static,
+{
+    /// Creates a new stub Schema Registry Service.
+    pub fn new(
+        application_context: ApplicationContext,
+        client: C,
+        output_directory_manager: &OutputDirectoryManager,
+    ) -> Self {
+        log::info!("Schema Registry Stub Service created");
+
+        Self {
+            schemas: Arc::new(Mutex::new(HashMap::new())),
+            get_command_executor: service_gen::GetCommandExecutor::new(
+                application_context.clone(),
+                client.clone(),
+                &CommandOptionsBuilder::default()
+                    .build()
+                    .expect("Default command options should be valid"),
+            ),
+            put_command_executor: service_gen::PutCommandExecutor::new(
+                application_context,
+                client,
+                &CommandOptionsBuilder::default()
+                    .build()
+                    .expect("Default command options should be valid"),
+            ),
+            service_output_manager: output_directory_manager
+                .create_new_service_output_manager(SERVICE_NAME),
+        }
+    }
+
+    /// Runs the Schema Registry stub service.
+    pub async fn run(self) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        let get_schema_runner_handle = tokio::spawn(Self::get_schema_runner(
+            self.get_command_executor,
+            self.schemas.clone(),
+        ));
+        let put_schema_runner_handle = tokio::spawn(Self::put_schema_runner(
+            self.put_command_executor,
+            self.schemas,
+            self.service_output_manager,
+        ));
+
+        tokio::select! {
+            r1 = get_schema_runner_handle => {
+                if let Err(e) = r1 {
+                    log::error!("Error in get_schema_runner: {:?}", e);
+                    return Err(Box::<dyn std::error::Error + Send + Sync>::from(e));
+                }
+            },
+            r2 = put_schema_runner_handle => {
+                if let Err(e) = r2 {
+                    log::error!("Error in put_schema_runner: {:?}", e);
+                    return Err(Box::<dyn std::error::Error + Send + Sync>::from(e));
+                }
+            }
+        };
+
+        Ok(())
+    }
+
+    async fn get_schema_runner(
+        mut get_command_executor: service_gen::GetCommandExecutor<C>,
+        schemas: Arc<Mutex<HashMap<String, BTreeSet<Schema>>>>,
+    ) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
does this need to be the Box error - it looks like only an AIOProtocolError could be returned for either fn